### ISSUE TITLE
Improving the traffic light system for colour blindness

### ISSUE THREAD

**Issue by K-Meech** [UCL-ARC/python-tooling#526]
### Which Section of the Documentation Needs Improving?

The sections involving red/orange/green traffic lights such as the home page / pages for linting, packaging etc..

### What Can be Improved About This Section

The traffic light system provides an easy to understand ranking of the different options, but presents some issues for people with colour blindness. The most common type of colour blindness (red / green) would make it difficult for people to distinguish the red/green traffic lights on colour alone, and other colour blindness types could make it difficult to distinguish all of them.

### How to Improve This Section

In the tables of ranking ([for example the linting one](https://github-pages.arc.ucl.ac.uk/python-tooling/pages/linting.html)) - would it be possible to provide another way to distinguish the categories? For example, text indicating the different groups, or potentially having the traffic lights as different shapes (like a green tick, red cross...).

**Comment by samcunliffe** [comment]
I pinged [UCL's Digital Accessibility](https://www.ucl.ac.uk/isd/services/digital-accessibility-services/digital-accessibility-team) team. They are going to take a general look at our pages, but in the meantime, they had some specific, signposted suggestions for this.

> To answer specifically about the traffic-light indicators, the most accessible approach would be to include descriptive text as well, as best practice is to avoid having colour/shape/size as the sole means of conveying information, and using plain text is the safest way to ensure screen readers can also access it. 
> [...]
>
> Our team has published [guidance on visuals and use of colour](https://www.ucl.ac.uk/isd/services/digital-accessibility-services/creating-accessible-content/visuals-and-use-colour), which discusses what to avoid in colour schemes, but I'll just drop in a few more links about colour and accessibility with specific suggestions included:
> 
> * Visualising Data has several ideas (including [alternative colour schemes to RAG](https://visualisingdata.com/2019/08/five-ways-to-design-for-red-green-colour-blindness/) on designing data visualisations for colour-blindness
  > * UX Collective suggests [using contrast and/or colour temperature to distinguish between reds and greens](https://uxdesign.cc/beautiful-accessible-traffic-light-colors-b2b14a102a38)

In particular the second link has a few nice images.

Here's a good illustration of the problem:

![](https://miro.medium.com/v2/resize:fit:720/format:webp/1*Q0pWIjaLXFnyMk4h06gXhw.png)


Here's a nice option:

![](https://miro.medium.com/v2/resize:fit:720/format:webp/1*SyI0gJll16MiM2DiAemNEA.png)


----

So a couple of options spring to mind...

1. Stick with emoji only but make them not be traffic light colours: ✅ ➖ ❌
2. Stick with emoji and traffic light colours but add shape: 💚🟠⛔️
3. Another set of emoji that are not traffic lights: 🙂😐☹️ or 👍👌 👎
4. Go with something like above: colours in the background of cell of the table, but also include a word describing our opinion: "Best", "Good", "Avoid"


Though technically the most fiddly (we'd have to do some CSS 😱 ) I wonder if option 4 is the best because we can use a word in plaintext

> best practice is to avoid having colour/shape/size as the sole means of conveying information

And for us "amber" is actually an OK tool:

> We don’t discourage using this, but it may duplicate functionality of a green tool.

So it's more of a "Best", "Good", "Avoid" kinda situation.

**Comment by samcunliffe** [comment]
Option 4 may not even be [that hard](https://just-the-docs.com/docs/ui-components/labels/#labels). But the rendered website would look differently to the .md files on GitHub (if anyone browses those).

**Comment by samcunliffe** [comment]
I've been playing around with what JustTheDocs already provides. 

Unfortunately [labels](https://just-the-docs.com/docs/ui-components/labels/) which look really nice, need a newline so can't work in tables.

Here's a go with some custom HTML span, but... it doesn't play nicely with themes like I'd have hoped...

https://github.com/user-attachments/assets/d3a5e54a-d4f7-4f82-a7cb-a9c6dafa3c59

**Comment by paddyroddy** [comment]
> Though technically the most fiddly (we'd have to do some CSS 😱 ) I wonder if option 4 is the best because we can use a word in plaintext

Why is that the best? I would prefer emojis and would work better in dark mode

**Comment by samcunliffe** [comment]
Well, "Best" == most accessible, because that's what the accessibility team said...

> the most accessible approach would be to include descriptive text 

But I think the point is: there's a plaintext thing that will appear if all emoji fails to render or can't be handled. (Like in screen readers for sight-impaired people)

**Comment by samcunliffe** [comment]
@all-contributors please add @K-Meech for bug, a11y, review
